scipy/interpolate/fitpack2.py

```diff
@@ -1380,6 +1433,19 @@ class SmoothSphereBivariateSpline(SphereBivariateSpline):
     """
 
     def __init__(self, theta, phi, r, w=None, s=0., eps=1E-16):
+
+        # input validation
+        if np.any(theta < 0.0) or np.any(np.pi < theta):
+            raise ValueError('theta should be between [0, pi]')
+        if np.any(phi < 0.0) or np.any(2.0 * np.pi < phi):
```

Comment:
I think these two intervals need double-checking: it's kind of odd to not be able to give `0` or `2*pi` as input. I suspect these are half-open intervals and the docstring is incorrect. And above you explicitly say "open interval", however the `< 0.0` and `> pi` checks imply a _closed_ interval.

In the `LSQSphereBivariateSpline` below you did change to `[0, 2pi]` (closed). 

Could you make this consistent?

Reply:
And for `RectSphereBivariateSpline` it's half-open ....

Reply:
I agree your comment. I did some tests for FITPACK, it looks like that FITPACK can accept closed interval. However, I think 360 degree interval should be half interval because 0 and 2pi amd -pi and pi is same degree. I made all intervals consistent. PTAL.

Reply:
I agree, that sounds good - then it's both consistent and backwards compatible.